### PR TITLE
Form builder

### PR DESCRIPTION
**Pull request by Codemax999**

### CODE REVIEW

File: Fenner_William-DFBT/js/index.js [hunk 1]
```diff
@@ -4,10 +4,15 @@
 
 function form_builder(attach, ...form_elements){
```
Comment (Codemax999) [inline Fenner_William-DFBT/js/index.js:5]: Why not use es6 arrow function? `const form_builder = (...) => {}`

File: Fenner_William-DFBT/js/index.js [hunk 2]
```diff
@@ -4,10 +4,15 @@
...
 
+    //Create a form element
     var form = document.createElement('form');
 
+    //Loop through all form elements from an array
     for(var i=0; i<form_elements.length; i++){
```
Comment (Codemax999) [inline Fenner_William-DFBT/js/index.js:11]: Later on in this loop the variables with only single characters become confusing, maybe instead use descriptive names for the variables.

File: Fenner_William-DFBT/js/index.js [hunk 3]
```diff
@@ -4,10 +4,15 @@
...
 
+        //Create an element based on the array input.
+        //input will create a text based input field
+        //button will create a button based on the type specified, etc
         if(form_elements[i][0] == "input"){
```
Comment (Codemax999) [inline Fenner_William-DFBT/js/index.js:16]: Consider a series of ternary operators to reduce code.`form.innerHTML += form_elements[i][0] === "label" ? someElement : ""`

File: Fenner_William-DFBT/js/index.js [hunk 4]
```diff
@@ -4,10 +4,15 @@
...
             form.innerHTML += "<"+form_elements[i][0]+" type='"+form_elements[i][1]+"' />";
```
Comment (Codemax999) [inline Fenner_William-DFBT/js/index.js:17]: You can simplify your string concatenation with the `` symbols: `<${form_elements[i][0]} type="${form_elements[i][1]}" />`

File: Fenner_William-DFBT/js/index.js [hunk 5]
```diff
@@ -21,19 +26,21 @@ function form_builder(attach, ...form_elements){
...
                 form.innerHTML += "<option value='"+form_elements[i][j]+"'>"+form_elements[i][j]+"</option>";
             }
 
-            form.innerHTML += "</ "+form_elements[i][0]+">";
+            form.innerHTML += "</select>";
         }
 
     }
 
     if(attach == ""){
```
Comment (Codemax999) [inline Fenner_William-DFBT/js/index.js:43]: The append to DOM approach is very cool, but project requires returning HTML string.

File: Fenner_William-DFBT/js/index.js (hunk 1 above)
Comment (wsf235) [reply Fenner_William-DFBT/js/index.js:5]: Honestly, I had Webstorm in JSX Harmony mode. I tried a couple different things at the time, but they returned errors on the syntax. But I see what you've done here so I have an idea on what revisions I want to do.

**Comment by wsf235** [comment]
At this point, I'm going to take the time for the rest of the day to review and recreate this form builder, now that I know where my faults lie. Thank you again, Cody.

**Pull request merged**